applications/OptimizationApplication/python_scripts/algorithms/algorithm_steepest_descent.py

```diff
@@ -1,129 +1,147 @@
+
+        self.__convergence_criteria = CreateConvergenceCriteria(settings["conv_settings"], self._optimization_problem)
+        self.__line_search_method = CreateLineSearch(settings["line_search"], self._optimization_problem)
+
+        self.__objective = StandardizedObjective(parameters["objective"], self.master_control, self._optimization_problem)
+        self.__control_field = None
+
+        self.__obj_val = None
+        self.opt_iter = None
+
+    def GetMinimumBufferSize(self) -> int:
+        return 2
+
+    def Check(self):
+        pass
+        
+    def Initialize(self):
+        self.converged = False
+        self.__obj_val = None
+
+        CallOnAll(self.master_control.GetListOfControls(), Control.Initialize)
+        self.__objective.GetReponse().Initialize()
+        self.__objective.Initialize()
+        self.__objective.Check()
+        self.__control_field = self.master_control.GetControlField() # GetInitialControlFields() later
+
+    def Finalize(self):
+        pass
+    
+    def ComputeSearchDirection(self, obj_grad) -> KratosOA.ContainerExpression.CollectiveExpressions:
+        return obj_grad * -1.0
+    
+    def GetCurrentObjValue(self) -> float:
+        return self.__obj_val
+    
+    def GetCurrentControlField(self):
+        return self.__control_field
+
+    def SolveOptimizationProblem(self) -> bool:
+        self.Initialize()
+        with TimeLogger("Solve Optimization problem", "Start", "End"):
+            self.Solve()
+        return self.converged
+    
+    def Solve(self):
+        algorithm_data = ComponentDataView("algorithm", self._optimization_problem)
+        while not self.converged:
+            print("")
+            with TimeLogger("Optimization", f" Start Iteration {self._optimization_problem.GetStep()}", f"End Iteration {self._optimization_problem.GetStep()}"):
+
+                with TimeLogger("Calculate objective value", "Start", "End"):
+                    self.__obj_val = self.__objective.CalculateStandardizedValue(self.__control_field) 
+                    algorithm_data.GetBufferedData()["std_obj_value"] = self.__obj_val
+                    print(self.__objective.GetInfo())
+
+                with TimeLogger("Calculate gradient", "Start", "End"):
+                    obj_grad = self.__objective.CalculateStandardizedGradient()
+                
+                with TimeLogger("Calculate design update", "Start", "End"):
+                    search_direction = self.ComputeSearchDirection(obj_grad)
+                    algorithm_data.GetBufferedData()["search_direction"] = search_direction
+
+                    for control in self.master_control.GetListOfControls():
+                        control_data_storage = ComponentDataView(control, self._optimization_problem)
+                        control_data_storage.GetUnBufferedData().SetValue("search_direction", search_direction.Clone(), overwrite=True)
```

Comment:
This is not longer required since it is already added to the `OptimizationProblem` in line 119.